### ISSUE TITLE
Scrape into sensor attribute a data table row/column  

### ISSUE THREAD

**Issue by wigster** [danieldotnl/ha-multiscrape#42]
**Is your feature request related to a problem? Please describe.**
I would like to scrape e.g. the forecast for temperature for the next 24 hours as an attribute of my temperature sensor (also being scraped). In the local weather service I use, this is presented as a 2-row table with 24 columns with pairs of values for time and temperature. 

See the detailed link to my use-case in [https://community.home-assistant.io/t/scrape-sensor-improved-scraping-multiple-values/218350/113?u=wigster](https://community.home-assistant.io/t/scrape-sensor-improved-scraping-multiple-values/218350/113?u=wigster).

A CSS selector which picks the whole data row, when processed by HA-MultiScrape returns only the first string. 

This sort of thing goes a little against HA philosophy, which is constructed to record a time series of data for the sensor in real-time. But I think in any situation with data about the future (e.g. weather forecast, tides, astronomical events, sporting events, TV schedules), it seems to me an ability to load a table of data would be useful.

**Describe the solution you'd like**
Ideally, it would be possible to return the data row as a sensor attribute in the form of a list. One would obtain this using some alternative version of value_template. It would be very helpful if the same transformation in this value_template (or one using the data item index as a variable) could be applied to each of the data items, e.g. converting the units of all of them, or manipulating the string in the same manner. 

In principle, performing statistical/mathematical operations on the list and returning just a single item could also be useful: e.g. picking the maximum expected tide from the list of the next 10, or returning tomorrow's low temperature.

**Describe alternatives you've considered**

Currently, it is possible to create separate attributes, one for each data point, but this is very inefficient and makes difficult further manipulation of these data.

**Additional context**

My aim is to use these data to plot the forecast in a lovelace card, using the ApexChart data generator feature
[https://github.com/RomRider/apexcharts-card#data_generator-option](https://github.com/RomRider/apexcharts-card#data_generator-option), which works easily with a particular list format for the attributes.

**Comment by danieldotnl** [comment]
Let's share the work :) If I provide you with a csv, could you turn it into your desired format with the value template?
E.g.
`16 °C,16 °C,15 °C,14 °C,14 °C,14 °C,13 °C,13 °C,14 °C,16 °C,18 °C,19 °C,20 °C,21 °C,22 °C,23 °C,23 °C,23 °C,23 °C,21 °C,20 °C,19 °C,19 °C,18 °C,17 °C`

**Comment by wigster** [comment]
You mean something like this?
```yaml
{% set value = "16 °C,16 °C,15 °C,14 °C,14 °C,14 °C,13 °C,13 °C,14 °C,16 °C,18 °C,19 °C,20 °C,21 °C,22 °C,23 °C,23 °C,23 °C,23 °C,21 °C,20 °C,19 °C,19 °C,18 °C,17 °C" -%}
{%-set value = value.split(",")-%}
{%-set out = namespace(bla=[])-%}
{%-for x in value-%}
  {%-set out.bla  = out.bla + [x[0:-2]|float]-%}
{%-endfor-%}
{{out.bla}}
```

**Comment by danieldotnl** [comment]
Almost! I meant the format you wanted to have:

  - 16
  - 16
  - 16
  - 14
  etc.

**Comment by wigster** [comment]
Yes, basically I would like to be able to get an attribute which would be a block style list of whatever data row/column I scrape, with some option to apply a filter/template to each data item. So here, for example, I need to remove the °C and make it into a float, but it could be different (e.g. a time what I would like to convert to timestamp or something like that). So in this example, something like:

```
TempForecast:
  - 16.0
  - 16.0
  - 15.0
  ...
```

**Comment by danieldotnl** [comment]
Could you try pre-release 5.2.0? It will provide you with the csv as mentioned above.
Add this to your configuration:
`select_list: ".day-hour-ext .h-100 .font-weight-bold"`
and remove the normal `select`

**Comment by wigster** [comment]
Wow, great. This is what I've done:
```
      attributes:
        - name: "Forecast Time"
          select_list: ".day-hour-ext.flex-column .day-hour-time"
          value_template: |
            {%-set value = value.split(",")-%}
            {%for x in value%}
            - {{x}}
            {%-endfor-%}
        - name: "Forecast Temperature"
          select_list: ".day-hour-ext .h-100 .font-weight-bold"
          value_template: |
            {%-set value = value.split(",")-%}
            {%for x in value%}
            - {{x[0:-2]|float}}
            {%-endfor-%}
```
And it gives me
```
forecast_time: |-
  - 18:00
  - 19:00
  - 20:00
  - 21:00
```
and then doing `{{state_attr('sensor.inpo_praha_temperature','forecast_time')}}` gives me the list. So I think this works exactly as I wanted it. Amazing!

Now to figure out the javascript to actually plot this...

Thanks a lot!

**Comment by danieldotnl** [comment]
You are welcome, good to hear it works out well! It would be great if you can post your findings and the sample value template on the forum as well!

Regarding your question: i believe the output is just a string by default but after converting it to a float, you could do calculations on the values. 
I think you can use entity_ids in an automation if templates are supported there. Probably better to ask this in a new thread on the forum as I believe it's not just related to multiscrape?

**Comment by Madelena** [comment]
Hello, I have a similar question. I am attempting to scrape a list of entries, but it's in JSON instead of HTML/XML. Since `select_list` wouldn't work on JSON, how can I make an array of dictionaries?

```
- name: NOAA Space Weather - Alerts
  resource: https://services.swpc.noaa.gov/products/alerts.json
  sensor:
    - unique_id: noaa_space_weather_alerts
      name: NOAA Space Weather - Alerts
      value_template: '{{ value_json[0]["product_id"] }}'
      attributes:
        - name: Entries
          value_template: |
            {% set temp = '' %}
            {% for message in value_json %}
            - product_id: {{ message.product_id }}
              issue_datetime: {{ message.issue_datetime }}
              message: {{ message.message }}
            {% endfor %}
```

which would output:

```
entries: - product_id: K05A
  issue_datetime: 2022-05-28 09:00:34.343
  message: Space Weather Message Code: ALTK05...
- product_id: K05W
  issue_datetime: 2022-05-28 05:47:24.707
  message: Space Weather Message Code: WARK05...
```

HA doesn't seem to be able to read it when I use the template `{{ state_attr('sensor.noaa_space_weather_alerts', 'entries')[0].message }}`.

**Comment by wigster** [comment]
I think this is related to the issue I had [here](https://github.com/danieldotnl/ha-multiscrape/issues/114#issuecomment-1113665125) -- it's seemingly a result of an internal limitation in HA. I've ended up scraping the list of data into one sensor as a text string and then parsing that as JSON with a separate template sensor, but this won't work if your data exceeds the 255 character limit.

**Comment by danieldotnl** [comment]
Finally understood and found the real issue. Fixed in v6.7.3!
